src/SignalR/clients/java/signalr/build.gradle

```
@@ -108,3 +108,27 @@ task generatePOM {
 }
 
 task createPackage(dependsOn: [jar,sourceJar,javadocJar,generatePOM])
+
+task generateVersionClass {
+    inputs.property "version", project.version
+    outputs.dir "$buildDir/generated"
+    doFirst {
+        def versionFile = file("$buildDir/../src/main/java/com/microsoft/signalr/Version.java")
+        versionFile.parentFile.mkdirs()
+        versionFile.text =
+                """
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.signalr;
+
+public class Version {
```

Let's make this package-private, so that only code in the `com.microsoft.signalr` can see it i.e.:

```suggestion
class Version {
```